packages/@aws-cdk/aws-servicecatalog/lib/portfolio.ts

```diff
@@ -66,6 +69,22 @@ export interface IPortfolio extends cdk.IResource {
    * @param options Options for the initiate share
    */
   shareWithAccount(accountId: string, options?: PortfolioShareOptions): void;
+
+  /**
+   * Associate portfolio with the given product.
+   * @param product A service catalog produt.
+   */
+  addProduct(product: IProduct): void;
+
+  /**
+   * Associate Tag Options
+   */
+  addTagOptions(tagOptions: TagOption): void;
+
+  /**
+   * Add a Resource Update Constraint.
+   */
+  allowTagUpdates(product: IProduct, options?: TagUpdatesOptions): void;
```

Comment:
1. Updated this to just use the constraint language which is just `addResourceUpdateConstraint`.
2. Our guidance is to not surface extra resources to customers when we don't need to or to make them track more variables in their stacks.  Same reason why we don't surface L1 constraints and abstracted them away to be created via methods.

Reply:
> Our guidance was to not surface these resources to customers when we don't need to, trying to limit the variables in their stacks that they need to track. Same reason why we don't surface L1 constraints and abstracted them away to be created via methods.

But the current design goes against that guidance, no?

Right now, in order to add a constraint, I have to track two variables (`product`, and `portfolio`) in my Stack - and this includes all helper functions/methods and constructs that I create. I always have to pass 2 things everywhere.

If you introduced a "product in portfolio" concept, the customer would only have a single variable to track, and pass.

Reply:
Well there are methods that are only called on the portfolio like adding users or tag options for example, so users still would be tracking those variables.  Additionally the business logic around constraints means that users wouldn't be adding a large number of them anyway since they have singular instances or mutual exclusivity.  

We did consider surfacing the association as its own resource to end user but decided not to for these reasons, but also because it would be less disruptive to add an association resource than to take it away.